fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -132,9 +133,25 @@ public LogicalMetric make(String metricName, String dependentMetric) {
      * @param metricName  Name for the metric we're making
      * @param dependentMetrics  Metrics this metric depends on
      *
-     * @return The new logicalMetric
+     * @return the new logicalMetric
+     *
+     * @deprecated logical metric needs more config-richness to not just configure metric name, but also metric long
+     * name, description, etc. Use {@link #makeInner(LogicalMetricInfo, List)} instead.
+     */
+    @Deprecated
+    protected LogicalMetric makeInner(String metricName, List<String> dependentMetrics) {
+        return makeInner(new LogicalMetricInfo(metricName), dependentMetrics);
+    }
+
+    /**
+     * Delegated to for actually making the metric.
+     *
+     * @param logicalMetricInfo  Logical metric info provider
+     * @param dependentMetrics  Metrics this metric depends on
+     *
+     * @return the new logicalMetric
      */
-    protected abstract LogicalMetric makeInner(String metricName, List<String> dependentMetrics);
+    protected abstract LogicalMetric makeInner(LogicalMetricInfo logicalMetricInfo, List<String> dependentMetrics);
```

Comment:
We should give it a default implementation, if not other peoples metric maker will break because they don't have this `makeInner` implemented.